magento/pwa-studio
# [PWA-2133] Cart : fix for out of stock products already in cart

## Description
If your cart contains products that do not have the requested quantity available, the items aren't rendered for you to be able to remove them. This PR renders the items when there's an error message, and displays the error message above the item. It also changes the price summary column to distinguish critical errors, since we don't want to display 2 separate error messages.

## Related Issue







Closes [PWA-2133](https://jira.corp.magento.com/browse/PWA-2133)

## Acceptance





### Verification Stakeholders



## Verification Steps






#### Test scenario(s) for direct fix/feature

- Access the admin panel of the backend instance
- Determine a product (variant) that you'll add to cart and visit the PDP on the frontend and the product edit page in the backend
- On the frontend, add a couple random product
- Add qty 50 of the product you're testing to your cart
- Add a couple other products
- Visit cart page (there should be no error)
- Change the available quantity in the backend to 40
- Reload the cart page
- There should be an error message above the cart item that's unavailable. The price summary should be visible but the checkout button should be disabled

If you do not have access to the backend, you can add a large quantity (ex 80) in one browser, and another large quantity in another browser and pass the order on one of them. Then test that the product isn't available for the requested qty in the remaining cart.

## Breaking Changes (if any)
**Requires our custom package since we're fetching a data point on the cart GQL that isn't in core**

## Checklist
